src/main/webapp/javascript/UserAuth.js

```diff
@@ -0,0 +1,23 @@
+
+// To be called before any request that requires user be logged in.
+// Action: if user not logged in, redirects to login page
+// redirects to redirectUrl regardless.
+// Args: String url redirectUrl defaults to the page function was called form.
+function confirmUserAndRedirect(redirectUrl) {
+    if (!redirectUrl) {
```

Comment:
indentation should be two spaces
[JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html#formatting-block-indentation)